### ISSUE TITLE
The key error

### ISSUE THREAD

**Issue by 2319183863** [yl4579/PL-BERT#25]
May I ask why I reported this error? I used the default code and data set
![image](https://github.com/yl4579/PL-BERT/assets/49605018/d2304418-97d8-48c9-88b8-8f86e92b4820)

**Comment by 2319183863** [comment]
Thanks for your reply. So the id is ranges from 0 to #vocab size and token is from tokenizer encoding (transfo-xl-wt103) right?
![image](https://github.com/yl4579/PL-BERT/assets/49605018/967fc2e2-8a76-4b12-8103-357aa228575c)

**Comment by tekinek** [comment]
> Thanks for your reply. So the id is ranges from 0 to #vocab size and token is from tokenizer encoding (transfo-xl-wt103) right? ![image](https://private-user-images.githubusercontent.com/49605018/285904877-967fc2e2-8a76-4b12-8103-357aa228575c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExMDQ0NjcsIm5iZiI6MTcwMTEwNDE2NywicGF0aCI6Ii80OTYwNTAxOC8yODU5MDQ4NzctOTY3ZmMyZTItOGE3Ni00YjEyLTgxMDMtMzU3YWEyMjg1NzVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI3VDE2NTYwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQwNjI4NWM4Y2YwZTM5MmE0Yjg3MGM1YmI3ZTQ5NzJiMWUwODg4ZGViYzE4MjE0MjUzNDAwODYwOGUxMTdmYTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.uyDQq39hgJP8X8DP5TMP_yMnVM2fdwA_x5yf4naUyzw)

Yes, that is right.

**Comment by 2319183863** [comment]
> > Thanks for your reply. So the id is ranges from 0 to #vocab size and token is from tokenizer encoding (transfo-xl-wt103) right? ![image](https://private-user-images.githubusercontent.com/49605018/285904877-967fc2e2-8a76-4b12-8103-357aa228575c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExMDQ0NjcsIm5iZiI6MTcwMTEwNDE2NywicGF0aCI6Ii80OTYwNTAxOC8yODU5MDQ4NzctOTY3ZmMyZTItOGE3Ni00YjEyLTgxMDMtMzU3YWEyMjg1NzVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI3VDE2NTYwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQwNjI4NWM4Y2YwZTM5MmE0Yjg3MGM1YmI3ZTQ5NzJiMWUwODg4ZGViYzE4MjE0MjUzNDAwODYwOGUxMTdmYTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.uyDQq39hgJP8X8DP5TMP_yMnVM2fdwA_x5yf4naUyzw)
> 
> Yes, that is right.

Thank you.

**Comment by meriamOu** [comment]
hey It seems that tokenizer: "transfo-xl-wt103" is only for english, I am using for 'bert-base-multilingual-cased' for spanish. Any idea how to find out the special token ? (to change this line in the config file :word_separator: 3039 # token used for word separator ())
thank you

**Comment by yl4579** [comment]
@meriamOu I think there are lot of unused tokens, just pick one you like.

**Comment by meriamOu** [comment]
thank you so much for your prompt reply. To double check, a potential config file will be like this:
dataset_params:
    tokenizer: "bert-base-multilingual-cased"
    token_separator: "[unused10]" # token used for phoneme separator (space)
    token_mask: "[MASK]" # token used for phoneme mask  
    word_separator: 9  # token used for word separator 
    token_maps: "token_maps.pkl" # token map path

**Comment by yl4579** [comment]
Yeah I thin that would work, but don't use `[MASK]`, use a single letter for masked token instead.

**Comment by meriamOu** [comment]
Hey again, using bert-base-multilingual-cased on wiki.spanish dataset ended up with unique index list of length  around 6 millions (even after eliminating the duplicates ) . Hence creating token_map is taking much longer time.
Will a token_map of 6 millions be a problem for the training? Would you still use bert-base-multilingual-cased ? 

Thank you so much

**Comment by tekinek** [comment]
> Hey again, using bert-base-multilingual-cased on wiki.spanish dataset ended up with unique index list of length around 6 millions (even after eliminating the duplicates ) . Hence creating token_map is taking much longer time. Will a token_map of 6 millions be a problem for the training? Would you still use bert-base-multilingual-cased ?
> 
> Thank you so much

token map with 6m uniq items is too many for trianing. I my case, 500k tokens are already enough big to make the training impossible with 24G of GPU memory (even with a batch size of 16 and maximum text length of 256)

**Comment by meriamOu** [comment]
thank you so much for replying! How did you reduce the number of tokens then, or did you use another tokenizer?

**Comment by tekinek** [comment]
> thank you so much for replying! It seems that the "transfo-xl-wt103" is only for english and since it a word tokonizer the size of the token map is almost equal to the vocab size.
> 
> All other multi language tokenizer use subwords splitting which leads to smaller vocab size but the number of token for the dataset is much much bigger (6 millions for spanish dataset) as we have a list of numbers for every word. @yl4579 how should we adjust the token_map to accommodate sub word tokonizer in order to train on other languages?
> 
> Looking forward to hearing back from you

This was the suggestion for supporning subword training:
https://github.com/yl4579/PL-BERT/issues/23#issuecomment-1826536331
